VoxelSniperForge/src/main/java/com/github/kevindagame/voxelsniperforge/world/ForgeWorld.java

```diff
@@ -152,8 +160,62 @@ public void regenerateChunk(int x, int z) {
+    public List<BrushOperation> generateTree(BaseLocation location, VoxelTreeType treeType, boolean updateBlocks) {
+        if (treeType.isSupported()) {
+            BlockPos pos = ForgeLocation.toForgeBlockPos(location);
+            BlockStateListPopulator populator = new BlockStateListPopulator(this);
+            boolean result = this.generateTree(populator, this.level.getChunkSource().getGenerator(), pos, new RandomSourceWrapper(RANDOM), treeType);
+            populator.refreshTiles();
+            if (!result) return null;
+
+            List<BrushOperation> ops = new ArrayList<>();
+            for (IBlockState newState : populator.getList()) {
+                var block = newState.getBlock();
+                var oldState = newState.getBlock().getState();
+                ops.add(new BlockStateOperation(new BaseLocation(this, block.getX(), block.getY(), block.getZ()), oldState, newState, true, true));
+                if (updateBlocks) {
+                    newState.update(true, true);
+                }
+            }
+            return ops;
+        }
+        return null;
+    }
+
+    private boolean generateTree(WorldGenLevel access, ChunkGenerator chunkGenerator, BlockPos pos, RandomSource random, VoxelTreeType treeType) {
+        ResourceKey<ConfiguredFeature<?, ?>> gen;
+        switch (treeType) {
+            case TREE -> gen = TreeFeatures.OAK;
+            case BIG_TREE -> gen = TreeFeatures.FANCY_OAK;
+            case REDWOOD -> gen = TreeFeatures.SPRUCE;
+            case TALL_REDWOOD -> gen = TreeFeatures.PINE;
+            case BIRCH -> gen = TreeFeatures.BIRCH;
+            case JUNGLE -> gen = TreeFeatures.MEGA_JUNGLE_TREE;
+            case SMALL_JUNGLE -> gen = TreeFeatures.JUNGLE_TREE_NO_VINE;
+            case COCOA_TREE -> gen = TreeFeatures.JUNGLE_TREE;
+            case JUNGLE_BUSH -> gen = TreeFeatures.JUNGLE_BUSH;
+            case RED_MUSHROOM -> gen = TreeFeatures.HUGE_RED_MUSHROOM;
+            case BROWN_MUSHROOM -> gen = TreeFeatures.HUGE_BROWN_MUSHROOM;
+            case SWAMP -> gen = TreeFeatures.SWAMP_OAK;
+            case ACACIA -> gen = TreeFeatures.ACACIA;
+            case DARK_OAK -> gen = TreeFeatures.DARK_OAK;
+            case MEGA_REDWOOD -> gen = TreeFeatures.MEGA_PINE;
+            case TALL_BIRCH -> gen = TreeFeatures.SUPER_BIRCH_BEES_0002;
+            case CHORUS_PLANT -> {
+                ChorusFlowerBlock.generatePlant(access, pos, random, 8);
+                return true;
+            }
+            case CRIMSON_FUNGUS -> gen = TreeFeatures.CRIMSON_FUNGUS_PLANTED;
+            case WARPED_FUNGUS -> gen = TreeFeatures.WARPED_FUNGUS_PLANTED;
+            case AZALEA -> gen = TreeFeatures.AZALEA_TREE;
+            case MANGROVE -> gen = TreeFeatures.MANGROVE;
+            case TALL_MANGROVE -> gen = TreeFeatures.TALL_MANGROVE;
+            case CHERRY -> gen = TreeFeatures.CHERRY;
+            default -> gen = TreeFeatures.OAK;
```

Comment:
This is troublesome, since This enum is leaving the codebase soon. (already removed in remove-enums

Reply:
I know, need to look for a better way

Reply:
Could you maybe do something like TREEFEATURES.valueOf(treeType.getKey().toUpperCase(Locale.ROOT))?

Reply:
yeah, something like that, there is a registry, but the names don't match with the names of our Treetype..

Reply:
should be much better now